app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,11 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts.includes(:comments)
+  end
 
-  def show; end
+  def show
+    @post = Post.includes(:comments).find(params[:id])
+    @user = User.find(params[:user_id])
+  end
 end
```

Comment:
[OPTIONAL] - This requirement is marked optional but highly recommended

app/model/post.rb

```ruby
class Post < ApplicationRecord
  validates :title, presence: true, length: { maximum: 250 }
  validates :comments_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
  validates :likes_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
end
```

I would suggest that you also create all necessary associations between models and all the necessary methods

associations
- A post belong to an author
- A post has many comments
- A post have many likes

Methods
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
```ruby
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :likes
  has_many :comments
  after_save :update_posts_counter

  validates :title, presence: true, length: { maximum: 250 }
  validates :comments_counter, numericality: true
  validates :likes_counter, numericality: true

  def update_posts_counter
    author.increment!(:posts_counter)
  end

  def five_recent_comments
    comments.order('created_at Desc').last(5)
  end
end
```

Please check here for [more info](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md)